lib/python/src/bailo/core/agent.py

```diff
@@ -95,14 +97,30 @@ def delete(self, *args, **kwargs):
 class TokenAgent(Agent):
     def __init__(
         self,
-        access_key: str,
-        secret_key: str,
+        access_key: str | None = None,
+        secret_key: str | None = None,
+        save_tokens: bool = True,
     ):
         """Initiate an agent for API token authentication.
 
         :param access_key: Access key
         :param secret_key: Secret key
+        :param save_tokens: Save tokens as environment variables, defaults to True
         """
+        super().__init__()
+
+        if access_key is None or secret_key is None:
+            try:
+                access_key = os.environ["BAILO_ACCESS_KEY"]
+                secret_key = os.environ["BAILO_SECRET_KEY"]
+            except:
+                access_key = getpass.getpass("BAILO ACCESS KEY ")
+                secret_key = getpass.getpass("BAILO SECRET KEY ")
```

Comment:
Just be aware this input will not work for some environments. Such as emulators or pseudo-terminals.